### DISCUSSION TITLE
lua插件问题

### DISCUSSION THREAD

**Discussion by rose8928** [opensumi/core#2011] [Q&A]
请问opensumi的对lua的插件功能兼容的全吗？我使用了vscode:     [ {
        "name": "lua",
        "version": "1.60.2"
      }],
和    "sumneko": [
      {
        "name": "lua",
        "version": "2.5.3"
      }
    ],
这两个插件，输入前几个字母，有函数、属性等提示，但是无法转到定义。vscode的sumneko插件是可以的，是我缺少哪个插件，还是版本的问题。基本上lua版本都试过了，我的版本是customVSCodeEngineVersion：1.60.2。
![lua1](https://user-images.githubusercontent.com/35814595/203714365-f0cc975e-8595-41d8-a40d-74b6e20c52f3.png)
![lua2](https://user-images.githubusercontent.com/35814595/203714381-1fbf171e-41c5-405a-93b1-993a201c05e6.png)

**Comment by erha19** [comment]
@rose8928 可以对照看一下你本地 VS Code 内插件市场，搜搜 `@builtin` 看一下是否缺少 language ... 相关语法服务插件

**Reply by rose8928** [reply]
我看到涉及到lua的这就两个插件，大佬能在opensumi-electron里给测试一下吗，我试了一下，opensumi-electron里好像也不行。

**Comment by erha19** [comment]
@rose8928 这个 language basic 插件需要装一下，同时也发现了内核版本的问题，你目前安装的 lua 插件对于内核版本要求应该是 `^1.67.0`, 需要你手动配置一下内核版本，或降级使用旧版本插件
<img width="406" alt="image" src="https://user-images.githubusercontent.com/9823838/203720879-69030063-7d55-4d1f-acb9-a6d903e5b357.png">
![2022-11-24 16 18 47](https://user-images.githubusercontent.com/9823838/203729506-ddcb9625-b87c-4980-aa16-bbb67f63d747.png)

**Reply by rose8928** [reply]
vscode.lua就是lua language basic这个插件吧,我已经安装了， "sumneko": [
{
"name": "lua",
"version": "2.5.3"
}这个插件也降低版本了，用的2..4.2，好像没问题，不报这个错。
应该是一共安装了两个插件，但是跳转到定义还是不行 ：）

**Reply by erha19** [reply]
@rose8928 这里的语法服务主要由 `sumneko.lua` 这个插件提供力，降级的话是否有效有待验证，暂时没时间看，你可以试试，目前 2.21 版本已支持绝大多数 1.69.0+ 版本 API，也可以升级后参考 https://github.com/opensumi/core/discussions/1885#discussioncomment-4025680 进行配置，将内核版本升高一点

**Reply by rose8928** [reply]
我想升级open-sumi,只要把package.json里的opensumi的依赖包改成2.21.0,然后重新pnpm install就可以了吧

**Reply by erha19** [reply]
@rose8928 一般是的，如果遇到问题可以参考 BreakingChanges 文档看一下对应修改
中文：https://github.com/opensumi/core/wiki/%E9%9B%86%E6%88%90%E6%8E%A5%E5%85%A5%20Breaking-Changes
英文：https://github.com/opensumi/core/wiki/Breaking-Changes

**Reply by rose8928** [reply]
好的，谢谢
